### PR TITLE
8326611: Clean up vmTestbase/nsk/stress/stack tests

### PR DESCRIPTION
**Pull request by lmesnik**
The vmTestbase/nsk/stress/stack might be slightly refactored to improve reading. Mostly, simple refactoring.
Test logic was not changed, except removing verbose mode and enabling printing.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8326611](https://bugs.openjdk.org/browse/JDK-8326611): Clean up vmTestbase/nsk/stress/stack tests (**Enhancement** - P4)

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)
 * [Mikhailo Seledtsov](https://openjdk.org/census#mseledtsov) (@mseledts - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17993/head:pull/17993` \
`$ git checkout pull/17993`

Update a local copy of the PR: \
`$ git checkout pull/17993` \
`$ git pull https://git.openjdk.org/jdk.git pull/17993/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17993`

View PR using the GUI difftool: \
`$ git pr show -t 17993`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17993.diff">https://git.openjdk.org/jdk/pull/17993.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17993#issuecomment-1962622326)

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/stress/stack/stack009.java [hunk 1]
```diff
@@ -52,44 +52,29 @@
...
-    public static int run(String args[], PrintStream out) {
-        for (int depth = 100; ; depth += 100)
+        for (int depth = 100; ; depth += 100) {
             try {
                 recurse(depth);
-            } catch (Error error1) {
-                if (!(error1 instanceof StackOverflowError) &&
-                        !(error1 instanceof OutOfMemoryError))
-                    throw error1;
+            } catch (StackOverflowError | OutOfMemoryError error1)  {
```
Comment (turbanoff) [inline test/hotspot/jtreg/vmTestbase/nsk/stress/stack/stack009.java:60]:
```suggestion
            } catch (StackOverflowError | OutOfMemoryError error1) {
```

**Comment by lmesnik** [comment]
Tests moved into runtime/stack.

**Comment by lmesnik** [comment]
The '-eager' mode is always true so I just removed this parameter.

File: test/hotspot/jtreg/TEST.groups [hunk 2]
```diff
@@ -415,7 +415,8 @@ tier1_runtime = \
  -runtime/Unsafe/RangeCheck.java \
   sanity/ \
  -:tier1_runtime_appcds_exclude \
- -runtime/signal
+ -runtime/signal \
+ -runtime/stack
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/TEST.groups:419]: So what tier will run these now? Are these really stress tests?

Comment (coleenp) [reply test/hotspot/jtreg/TEST.groups:419]: Good question.  They are stress tests and not something we want to run in tier1.

**Review by coleenp — APPROVED** [review]
Thank you for moving these.

File: test/hotspot/jtreg/TEST.groups (hunk 2 above)
Comment (lmesnik) [reply test/hotspot/jtreg/TEST.groups:419]: These tests are executed in tier2 now. They are stress tests, trying to provoke SOE in different ways. They don't need a lot of resources but might take a long time which is not recommended for tier1 tests.

**Review by dholmes-ora — COMMENTED** [review]
Copyrights need updating in many of the files.

What is the `Terminator` that some of the tests were using but now are not?

Do we really have 19 different stackoverflow tests here? The differences in some of the tests seem inconsequential in terms of what they are testing e.g. 004 uses a static method and 005 uses a non-static method (based on the descriptions).

**Comment by lmesnik** [comment]
Thank you, I removed the Terminator.

I completely agree that it is needed to review tests and merge/remove some of them and might be rename . However, I would like to file another bug for this and don't mix it with the current plain clean up. The test removal need more explanations.

**Comment by dholmes-ora** [comment]
> Thank you, I removed the Terminator.

Okay, but what was it doing?

**Comment by lmesnik** [comment]
This class starts the guard thread which calls System.exit(95) after some time. It is controlled '-waittime' parameter which is also used in NSK stress tests.

**Review by coleenp — APPROVED** [review]
This continues to look really good.  There were several stack tests that were similar but would trigger StackOverflow in subtly different places, so there were (at the time) no duplicates.  Each had a bug fixed for it at the time. I remember getting segv in the 'instanceof' native code and some println calls which determinted how large to create the StackShadowPages.  There still seem to be some instanceof in the catches so that's good.  If there are true duplicates now, it would be good to remove them.

**Review by mseledts — APPROVED** [review]

**Pull request closed without merging**